code/game/objects/items/devices/radio/beacon.dm

```diff
@@ -11,6 +11,14 @@
 	var/code = "electronic"
 	var/functioning = TRUE
 
+/obj/item/radio/beacon/Initialize()
+	. = ..()
+	teleporter_beacons |= src
```

Comment:
Style is `global.teleporter_beacons` when referring to a globally scoped variable.
`|=` is unnecessary here (vs `+=`) as `Initialize()` is only called once ever.